dataStorage/storage.js

```diff
@@ -33,15 +33,21 @@ async function loadAsync(model, id) {
     
     // try to load from cache first
     let data = null;
-    let keyString = (id == null)? model.key: model.key + id;    
+    let keyString = (id == null)? model.key: model.key + id;
     if (model.cachePolicy == CachePolicy.Memory) {
         data = global.cache[keyString];
     }
     else if (model.cachePolicy == CachePolicy.AsyncStorage) {
         data = await loadFromOffilneStorageAsync(model.key, id);
-    }    
+        if (data && update) {
+            console.log("updating the data from cloud");
+            let updatedData = await loadFromCloudAsync(model, id, /*silentLoad*/ true);
+            console.log("silently save the updated cloud data to storage");
+            saveToOffilneStorageAsync(updatedData, model.key, id);
```

Comment:
do you need to check the error case? maybe we should skip save operation due to network failures?